src/linkify-element.js

```diff
@@ -32,8 +32,11 @@ function tokensToNodes(tokens, opts, doc) {
 
 	for (let i = 0; i < tokens.length; i++) {
 		let token = tokens[i];
+		let validated = options.resolve(opts.validate,
+			token.hasProtocol ? token.hasProtocol() : false,
```

Comment:
For example, if the tokenizer already knows how to handle protocols, it is redundant to have to rehandle them with a regex (or something like that).